src/folderview.cpp

```diff
@@ -182,6 +182,27 @@ void FolderViewListView::mouseDoubleClickEvent(QMouseEvent* event) {
     activationAllowed_ = activationWasAllowed;
 }
 
+QModelIndex FolderViewListView::moveCursor(CursorAction cursorAction, Qt::KeyboardModifiers modifiers) {
+    QAbstractItemModel* model_ = model();
+
+    if(!model_ || !currentIndex().isValid()) {
+        return QModelIndex();
```

Comment:
Here, `return QModelIndex();` should be replaced by `return QListView::moveCursor(cursorAction, modifiers);`, otherwise there will be no action after opening a folder.

Preferably, you could put your code under the condition `if(model_ && currentIndex().isValid())` and have only one instance of `return QListView::moveCursor(cursorAction, modifiers);` at the end (outside the condition).